src/CalendarDate.ts

```diff
@@ -119,6 +119,30 @@ export class CalendarDate {
       .padStart(2, '0')}-${this.day.toString().padStart(2, '0')}`;
   }
 
+  /**
+   * Returns a string representation formatted according to the specified format string.
+   * Supports the following Tokens:
+   *
+   * - **yyyy**: four digit year
+   * - **yy**: two digit year
+   * - **y**: year without padding
+   * - **MM**: month padded to 2 digits
+   * - **M**: month without padding
+   * - **dd**: day padded to 2 digits
+   * - **d**: day without padding
+   *
+   */
+  toFormat(pattern: string): string {
+    return pattern
+      .replace(/yyyy/g, this.year.toString().padStart(4, '0'))
+      .replace(/yy/g, this.year.toString().slice(-2).padStart(2, '0'))
+      .replace(/y/g, this.year.toString())
+      .replace(/MM/g, (this.month + 1).toString().padStart(2, '0'))
```

Comment:
What was the reason to use M as capital letter for formatting month?

Reply:
m is reserved for minutes